### DISCUSSION TITLE
Dynamic authentication on REST API

### DISCUSSION THREAD

**Discussion by banzai69X** [Budibase/budibase#4771] [Help]
Hi everyone,

I've a database BuddiDB and inside a table nammed "devices" with a column "apikey" and other column "secretapi"
I want to create a get request with REST api to check version of each device so i need to launch this query with an apikey et secretapi diferent for each device.

So the idea would be have an automation who will call for each device my get query with params to authenticate with a header Authorization.

How can I do this please ?
Thanks all

**Comment by Topener** [comment]
Hi, unfortunately, this is not (yet) something we have support for. We do have the ability to consume API's but this is through a data source and hardcoded authentication and headers.

I've logged your request in a feature request ticket #4774
